packages/nx/src/native/tui/app.rs

```diff
@@ -774,17 +939,442 @@ impl App {
+        tokio::spawn(async move {
+            let _ = tx.send(action);
+        });
+    }
+
+    fn recalculate_layout_areas(&mut self) {
+        if let Some(frame_area) = self.frame_area {
+            self.layout_areas = Some(self.layout_manager.calculate_layout(frame_area));
+        }
+    }
+
+    /// Checks if the current view has any visible output panes.
+    fn has_visible_panes(&self) -> bool {
+        self.pane_tasks.iter().any(|t| t.is_some())
+    }
+
+    /// Clears all output panes and resets their associated state.
+    fn clear_all_panes(&mut self) {
+        self.pane_tasks = [None, None];
+        self.update_focus(Focus::TaskList);
+        self.set_spacebar_mode(false, None);
+        self.dispatch_action(Action::UnpinAllTasks);
+    }
+
+    /// Toggles the visibility of the output pane for the currently selected task.
+    /// In spacebar mode, the output follows the task selection.
+    fn toggle_output_visibility(&mut self) {
+        self.task_list_hidden = false;
+        self.layout_manager
+            .set_task_list_visibility(TaskListVisibility::Visible);
+
+        // Extract task name first to end the immutable borrow
+        let task_name = match self
+            .selection_manager
+            .lock()
+            .unwrap()
+            .get_selected_task_name()
+        {
+            Some(name) => name.clone(),
+            None => return,
+        };
+
+        if self.has_visible_panes() {
+            self.clear_all_panes();
+            self.set_spacebar_mode(false, None);
+        } else {
+            // Show current task in pane 1 in spacebar mode
+            self.pane_tasks = [Some(task_name), None];
+            self.set_spacebar_mode(true, None);
+        }
+    }
+
+    fn set_spacebar_mode(
+        &mut self,
+        spacebar_mode: bool,
+        selection_mode_override: Option<SelectionMode>,
+    ) {
+        self.spacebar_mode = spacebar_mode;
+
+        let selection_mode = if spacebar_mode {
+            // When entering spacebar mode, we want to track by name by default
+            SelectionMode::TrackByName
+        } else {
+            // When exiting spacebar mode, we want to track by position by default
+            SelectionMode::TrackByPosition
+        };
+        self.selection_manager
+            .lock()
+            .unwrap()
+            .set_selection_mode(selection_mode_override.unwrap_or(selection_mode));
+
+        if spacebar_mode {
+            self.layout_manager
+                .set_pane_arrangement(PaneArrangement::Single);
+        } else {
+            self.layout_manager
+                .set_pane_arrangement(PaneArrangement::None);
+        }
+
+        // Recalculate the layout areas
+        self.recalculate_layout_areas();
+
+        // Ensure the pty instances get resized appropriately
+        let _ = self.debounce_pty_resize();
+
+        self.dispatch_action(Action::SetSpacebarMode(spacebar_mode));
+    }
+
+    fn focus_next(&mut self) {
```

Comment:
future: Maybe create a focus manager

This is a bunch of code which was moved from tasks_list.rs..

So the story so far seems to be. app.rs was too big.. split it out into tasks_list.rs... then now we're saying tasks_list component is too big... so we've moved this back into app.rs....

I think there's enough code handling focus here to justify it's own module?